codecov: Sanitize sent flags

leandro-lucarella-sociomantic opened the pull request:
A codecov flags should only have a restricted set of chars:

> flags must match pattern ^[\w\,]+$

But now they are send with whatever they have, causing sent reports to fail.

leandro-lucarella-sociomantic committed:
codecov: Properly sanitize sent flags

Only DMD versions was partially sanitized, but we actually need to
convert all strange chars (anything that is not `\w`), not just a few.

This commit also sanitizes other flags sent to avoid future problems.

Fixes #36.

leandro-lucarella-sociomantic commented:
Fix attached.

mihails-strasuns-sociomantic commented on bin/dlang/codecov:
```diff
@@ -42,9 +42,22 @@ git archive --format=tar HEAD | (cd "$tmp" && tar xf -)
+    flag="$(echo "$flag" | sed -r 's/\W/_/g')"
+    if test -z "$flags"
+    then
+        flags="-F $flag"
```
Protection against empty `$flag` while you are at it? :)

mihails-strasuns-sociomantic replied:
Ping @leandro-lucarella-sociomantic

leandro-lucarella-sociomantic replied:
Mmm, I guess, I wanted to add a prefix to all flags, like we have with dmd. Maybe I can do that so flags will show if it was built with an empty `$F` for example.

leandro-lucarella-sociomantic committed:
codecov: Add prefix to all sent flags

This makes sure we don't send empty flags and we can tell the kind of
flag is used, so there is no ambiguity if we introduce new flags.

leandro-lucarella-sociomantic commented:
Updated

mihails-strasuns-sociomantic commented:
@leandro-lucarella-sociomantic can you make a point release right away? ^_^
